src/org/intellij/erlang/quickfixes/ErlangCreateFunctionQuickFix.java

```diff
@@ -24,81 +25,104 @@
 import com.intellij.openapi.editor.Editor;
 import com.intellij.openapi.project.Project;
 import com.intellij.psi.PsiElement;
-import com.intellij.psi.util.PsiTreeUtil;
-import com.intellij.psi.util.PsiUtilBase;
-import com.intellij.util.Function;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.SmartPointerManager;
+import com.intellij.psi.SmartPsiElementPointer;
+import com.intellij.util.IncorrectOperationException;
 import com.intellij.util.containers.ContainerUtil;
-import org.intellij.erlang.psi.*;
-import org.intellij.erlang.refactoring.ErlangRefactoringUtil;
+import org.jetbrains.annotations.Nls;
 import org.jetbrains.annotations.NotNull;
 
-import java.util.ArrayList;
 import java.util.List;
 
-public class ErlangCreateFunctionQuickFix extends LocalQuickFixBase {
-  private final String myName;
-  private final int myArity;
+public class ErlangCreateFunctionQuickFix extends LocalQuickFixBase implements IntentionAction {
+  private final FunctionTextProvider<? extends PsiElement> myFunctionText;
+  private final String myFixMessage;
 
-  public ErlangCreateFunctionQuickFix(@NotNull String name, int arity) {
-    super("Create Function '" + name + "/" + arity + "'", "Erlang");
-    myName = name;
-    myArity = arity;
+  public ErlangCreateFunctionQuickFix(@NotNull PsiElement element, @NotNull String name, int arity,
+                                      @NotNull String fixMessage) {
```

Comment:
I've tried my best to explain to you, that you do not need a problem element in this case. Take a look at where it goes. You never use it!